File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
      // left = 0;
      //int sum = 0;
      //right for in range(nums)
      //sum+= nums[right]
      //while( sum >= target) // update min len, sum, move left pointer,

      //return if minLength inf, return 0 ow return minlen.


      int n = nums.Length;
      int left = 0;
      int sum = 0;
      int minLen = int.MaxValue; // minLen = inf.
       

      for (int right = 0; right < n; right++){
        sum += nums[right]; //sliding window.
        while(sum >= target){
          int currLen = right - left + 1;
          minLen = Math.Min(minLen, currLen); // currL = right - left + 1
          sum -= nums[left];
          left++;

        }
      }
      //we found the minLen of target that consecutively satsifies.

      return (minLen == int.MaxValue) ? 0 : minLen;



        
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_len = float("inf")

        left = 0
        cur_sum = 0
        for right in range(len(nums)):
            cur_sum += nums[right]
            while cur_sum >= target:
                cur_sum -= nums[left]

                min_len = min(min_len, right - left + 1)

                left += 1
        return min_len if min_len != float("inf") else 0
        






        










    #   left = 0
    #   window_sum = 0
    #  # result = []
    #   result = float('inf')
      


    #   for right in range(len(nums)):
    #     window_sum += nums[right]
    #     while window_sum >= target:
          
    #       window_sum -= nums[left]
          
    #       result = min(result, right - left + 1)
    #       left += 1
          
          
    #   return result if result != float('inf') else 0
          




          


      

